src/hpx_init.cpp

```diff
@@ -337,143 +348,182 @@ namespace hpx
             hpx::program_options::variables_map& vm,
             bool print_counters_locally)
         {
-            if (vm.count("hpx:list-counters")) {
-                // Print the names of all registered performance counters.
-                std::string option(vm["hpx:list-counters"].as<std::string>());
-                if (0 == std::string("minimal").find(option))
-                    rt.add_startup_function(&list_counter_names_minimal);
-                else if (0 == std::string("full").find(option))
-                    rt.add_startup_function(&list_counter_names_full);
-                else {
-                    std::string msg ("Invalid command line option value"
-                        "for --hpx:list-counters: ");
-                    msg += option;
-                    msg += ", allowed values are 'minimal' and 'full'";
-                    throw detail::command_line_error(msg.c_str());
+#if defined(HPX_HAVE_DISTRIBUTED_RUNTIME)
+            runtime_distributed* rtd =
+                dynamic_cast<hpx::runtime_distributed*>(&rt);
+            if (rtd != nullptr)
```

Comment:
I think an `assert` would be fine here. If the distributed runtime is compiled in, this has to succeed.